src/jws.c

```diff
@@ -313,13 +313,21 @@ static bool _cjose_jws_build_dig_hmac_sha(
     }
 
     // instantiate and initialize a new mac digest context
+#if (CJOSE_OPENSSL_11X)
+    ctx = HMAC_CTX_new();
+#else
     ctx = cjose_get_alloc()(sizeof(HMAC_CTX));
+#endif
     if (NULL == ctx)
     {
-        CJOSE_ERROR(err, CJOSE_ERR_CRYPTO);
+        CJOSE_ERROR(err, CJOSE_ERR_NO_MEMORY);
         goto _cjose_jws_build_dig_hmac_sha_cleanup;
     }
+
+#if (CJOSE_OPENSSL_11X)
```

Comment:
I think there should be a comment here about why an empty `#if` is really ok.  I can imagine this inadvertently being removed in the future without some indication of why it's there.